tests/test_recompile_ux.py

```diff
@@ -0,0 +1,92 @@
+from abc import ABC
+from abc import abstractmethod
+from typing import Any
+from typing import List
+
+import torch
+
+import torchdynamo
+import torchdynamo.config
+import torchdynamo.testing
+
+torchdynamo.config.debug = False
+from torchdynamo.utils import counters
+
+class RecompileUxTests(torchdynamo.testing.TestCase):
+    # TODO(whc) dynamo actualy recompiles one more time than the cache limit
```

Comment:
I'd expect at least one extra compile to come from a graph break.

You should get one graph containing only `torch.randint(low=0, high=1000, size=())`, which gets compiled once.

The limit is on a per-code-object bases.  And graph breaks create new code objects after the break.